### PR TITLE
fix: fix flaky e2e tests

### PR DESCRIPTION
**Pull request by britaniar**
### Description of your changes

Fixes #

I have: 
- change the name so of the cluster scoped resource so it doesn’t conflict with another 
- Fixed the eventually function variable 
- Ensure the correct namespace is called 

- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: test/e2e/placement_selecting_resources_test.go [hunk 1]
```diff
@@ -483,7 +483,7 @@ var _ = Describe("validating CRP when adding resources in a matching namespace",
 			workResourcesPlacedActual := func() error {
 				return validateWorkNamespaceOnCluster(memberCluster, workNamespaceName)
 			}
-			Expect(workResourcesPlacedActual()).Should(Succeed(), "Failed to place work namespace %s on member cluster %s", workNamespaceName, memberCluster.ClusterName)
+			Eventually(workResourcesPlacedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to place work namespace %s on member cluster %s", workNamespaceName, memberCluster.ClusterName)
```
Comment (zhiying-lin) [inline test/e2e/placement_selecting_resources_test.go:486]: are you sure this test is flaky?

here it should use expect, cause crp has already applied the resources.
no need to use eventually.

File: test/e2e/placement_selecting_resources_test.go [hunk 2]
```diff
@@ -856,7 +855,7 @@ var _ = Describe("validating CRP when failed to apply resources", Ordered, func(
 
 var _ = Describe("validating CRP when placing cluster scope resource (other than namespace)", Ordered, func() {
 	crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
-	clusterRoleName := fmt.Sprintf("reader-%d", GinkgoParallelProcess())
+	clusterRoleName := fmt.Sprintf("test-cluserrole-%d", GinkgoParallelProcess())
```
Comment (zhiying-lin) [inline test/e2e/placement_selecting_resources_test.go:858]: will rename help to reduce the flakiness?

File: test/e2e/placement_selecting_resources_test.go [hunk 3]
```diff
@@ -1347,7 +1346,7 @@ func checkIfPlacedLargeSecretResourcesOnTargetMemberClusters(targetMemberCluster
 		memberCluster := targetMemberClusters[idx]
 
 		secretsPlacedActual := secretsPlacedOnClusterActual(memberCluster)
-		Eventually(secretsPlacedActual(), largeEventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to place large secrets on member cluster %s", memberCluster.ClusterName)
+		Eventually(secretsPlacedActual, largeEventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to place large secrets on member cluster %s", memberCluster.ClusterName)
```
Comment (zhiying-lin) [inline test/e2e/placement_selecting_resources_test.go:1349]: good catch!

File: test/e2e/placement_selecting_resources_test.go (hunk 1 above)
Comment (britaniar) [reply test/e2e/placement_selecting_resources_test.go:486]: it failed in these two commits:
- [feat: property-based scheduling: enable the property-based scheduling experience #2695](https://github.com/Azure/fleet/actions/runs/8506422376/job/23296596391)
- [feat: apply overrides before creating the work #2705](https://github.com/Azure/fleet/actions/runs/8517373471/job/23327835302)

File: test/e2e/placement_selecting_resources_test.go (hunk 2 above)
Comment (britaniar) [reply test/e2e/placement_selecting_resources_test.go:858]: Will check again, but without the rename it was failing locally on my machine.

File: test/e2e/placement_selecting_resources_test.go (hunk 1 above)
Comment (zhiying-lin) [reply test/e2e/placement_selecting_resources_test.go:486]: let me investigate it, this one looks like a bug.

File: test/e2e/placement_selecting_resources_test.go [hunk 4]
```diff
@@ -428,10 +428,10 @@ var _ = Describe("validating CRP when cluster-scoped and namespace-scoped resour
 
 var _ = Describe("validating CRP when adding resources in a matching namespace", Ordered, func() {
 	crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+	ns := appNamespace()
```
Comment (ryanzhang-oss) [inline test/e2e/placement_selecting_resources_test.go:431]: why moving this out of the BeforeAll? Is there any difference?

**Review by zhiying-lin — APPROVED** [review]

**Pull request merged**